components/experimental/src/dimension/currency/long_compact_format.rs

```diff
@@ -0,0 +1,128 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::CurrencyCode;
+use crate::alloc::borrow::ToOwned;
+use crate::compactdecimal::CompactDecimalFormatter;
+use crate::dimension::provider::currency_patterns::CurrencyPatternsDataV1;
+use crate::dimension::provider::extended_currency::CurrencyExtendedDataV1;
+use fixed_decimal::SignedFixedDecimal;
+use icu_plurals::PluralRules;
+use writeable::Writeable;
+
+pub struct LongCompactFormattedCurrency<'l> {
+    pub(crate) signed_fixed_decimal: &'l SignedFixedDecimal,
+    // TODO: use this if the display name is not exist and make the extended data optional.
+    pub(crate) _currency_code: CurrencyCode,
+    pub(crate) extended: &'l CurrencyExtendedDataV1<'l>,
+    pub(crate) patterns: &'l CurrencyPatternsDataV1<'l>,
+    pub(crate) compact_decimal_formatter: &'l CompactDecimalFormatter,
+    pub(crate) plural_rules: &'l PluralRules,
+}
+
+writeable::impl_display_with_writeable!(LongCompactFormattedCurrency<'_>);
+
+impl Writeable for LongCompactFormattedCurrency<'_> {
+    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
+    where
+        W: core::fmt::Write + ?Sized,
+    {
+        let operands = self.signed_fixed_decimal.into();
+
+        let display_name = self.extended.display_names.get(operands, self.plural_rules);
+        let pattern = self.patterns.patterns.get(operands, self.plural_rules);
+
+        let formatted_value = self
+            .compact_decimal_formatter
+            .format_fixed_decimal(self.signed_fixed_decimal.to_owned());
```

Comment:
Nit: Avoid `to_owned()` here. `format_fixed_decimal` should be made to borrow instead of value if that's part of the problem.

Reply:
Okay, I am adding a TODO and filed a bug  to fix format_fixed_decimal (https://github.com/unicode-org/icu4x/issues/5881)

once I am fixing the bug, I will remove the `to_owned()` calling